lib/runner/variant-reassignment.js

```diff
@@ -289,15 +384,31 @@ export default class VariantReassignment {
     if (products) {
       const skuToProductMap = this._createSkuToProductMap(products)
       const matchingProducts = productDraftSkus.map(sku => skuToProductMap.get(sku))
-      return _.uniq(matchingProducts)
+
+      // when there is a new non existing variant in product draft we will get undefined
+      // in matchingProducts because there is no existing product so with _.compact we
+      // will remove undefined value and work only with variants which are existing on API
+      // and should be reassigned
+      return _.uniq(_.compact(matchingProducts))
     }
     return this.productService.getProductsBySkus(productDraftSkus)
   }
 
-  _deleteTransaction () {
+  _createNewProduct (product, productTypeId) {
+    product.productType.id = productTypeId
+
+    const projection = this.productService.transformProductToProjection(product)
+    projection.productType.id = productTypeId
+    return this.productService.createProduct(projection)
   }
 
-  _createNewProduct () {
+  async _ensureAnonymizedProductDraft (anonymizedProductDraft) {
```

Comment:
Better name: `ensureProductCreation`? It doesn't have to be anonymized.

Reply:
Updated 8e56cf3